pkg/util/test/util.go

```diff
@@ -1042,23 +1078,49 @@ func ConstructPxReleaseManifestURL(specGenURL string) (string, error) {
 	return u.String(), nil
 }
 
-func validateStorageClusterIsOnline(cluster *corev1.StorageCluster, timeout, interval time.Duration) (*corev1.StorageCluster, error) {
-	t := func() (interface{}, bool, error) {
+func validateStorageClusterInState(cluster *corev1.StorageCluster, status string) func() (interface{}, bool, error) {
+	return func() (interface{}, bool, error) {
 		cluster, err := operatorops.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
 		if err != nil {
 			return nil, true, fmt.Errorf("failed to get StorageCluster %s in %s, Err: %v", cluster.Name, cluster.Namespace, err)
 		}
 
-		if cluster.Status.Phase != "Online" {
+		if cluster.Status.Phase != status {
 			if cluster.Status.Phase == "" {
 				return nil, true, fmt.Errorf("failed to get cluster status")
 			}
 			return nil, true, fmt.Errorf("cluster state: %s", cluster.Status.Phase)
 		}
 		return cluster, false, nil
 	}
+}
 
-	out, err := task.DoRetryWithTimeout(t, timeout, interval)
+func validateAllStorageNodesInState(namespace string, status corev1.NodeConditionStatus) func() (interface{}, bool, error) {
+	return func() (interface{}, bool, error) {
+		// Get all StorageNodes
+		storageNodeList, err := operatorops.Instance().ListStorageNodes(namespace)
+		if err != nil {
+			return nil, true, fmt.Errorf("failed to list StorageNodes in %s, Err: %v", namespace, err)
+		}
+
+		for _, node := range storageNodeList.Items {
+			for _, cond := range node.Status.Conditions {
+				if cond.Type != corev1.NodeInitCondition {
```

Comment:
When the NodeInit condition fails, doesn't the node.Status.Phase change to 'Failed' too? I thought it did. If not, it seems like an issue in operator code. If yes, you can simply use phase instead of going through the conditions.

Reply:
I'll go ahead and confirm, this may have just been a case of me trying to find the right variable and not noticing "Phase" as being equivalent to "Status". I'll let you know either way.

Reply:
Discussed on Slack, will be assuming StorageNode status will be correct for now and have opened a bug as far as status not updating properly.